src/utils/hooks/useLocalStorage.tsx

```diff
@@ -0,0 +1,25 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { useState } from 'react';
+
+export const useLocalStorage = (key: string, initialValue: any) => {
+  const getValues = () => {
+    const storage = localStorage.getItem(key);
+
+    if (!storage) {
+      localStorage.setItem(key, JSON.stringify(initialValue));
+
+      return initialValue;
+    }
+
+    return JSON.parse(storage || '');
```

Comment:
This line can lead to an exception if the item from the local storage cannot be parsed into JSON. You should always make sure that the data you are trying to parse is valid JSON before calling JSON.parse.